kpet/data.py

```diff
@@ -329,6 +379,64 @@ def is_dir_valid(dir_path):
         """
         return os.path.isfile(dir_path + "/index.yaml")
 
+    def validate(self):
+        """
+        Check if any of the regexes are invalid.
```

Comment:
I think a more specific description is in order here. E.g. "Check patterns match possible targets only". And then, if you're not saying you're validating the whole object here, better be more specific with the name. E.g. name this function `validate_patterns`.